### PR TITLE
Limit git gc to when dserve is idle

### PR DESCRIPTION
**Pull request by blowery**
Git gc takes a fair bit of memory, so avoid trying to run it when the server is busy.

### CODE REVIEW

File: src/api.ts [hunk 1]
```diff
@@ -301,13 +293,15 @@ async function getRemoteBranches(): Promise< Map< string, string > > {
...
-		try {
-			await promisify( exec )( 'git gc', {
-				cwd: calypsoDir,
-			} );
-		} catch ( err ) {
-			l.error( { err }, 'git gc failed' );
+		// gc the repo if no builds are running
+		if ( pendingHashes.size === 0 ) {
+			try {
+				await promisify( exec )( 'git gc', {
```
Comment (sirreal) [inline src/api.ts]: I like to _be aggressive_.

```suggestion
				await promisify( exec )( 'git gc --prune=all --aggressive', {
```

Comment (sirreal) [inline src/api.ts]: The fact that we only log on error makes it tricky to know whether we're doing this without error or if we never do it.

Comment (blowery) [reply src/api.ts]: Are you sure? https://git-scm.com/docs/git-gc#_aggressive

> Not using this at all is the right trade-off for most users and their repositories.

Comment (blowery) [reply src/api.ts]: and we already do the prune as part of the fetch

Comment (blowery) [reply src/api.ts]: always logging just clogs up the logs. we do this pretty often. maybe too often.

**Pull request merged**